README.md

```diff
@@ -1,9 +1,75 @@
 # .NET Source Build Externals
 
-This repo contains the source that resides outside of the dotnet organization required to build .NET from source.
-Examples include Newtonsoft.Json and Application Insights for .NET. Git submodules are utilized to reference the external source.
-This repo contains the infrastructure to build these external repos within the .NET source build.
-See [dotnet/source-build](https://github.com/dotnet/source-build) for more details on .NET source build.
+This repo contains the source for components that resides outside of the dotnet
+organization required to build .NET from source. Examples include Newtonsoft.Json
+and Application Insights for .NET. Git submodules are utilized to reference the
+external source. This repo contains the infrastructure to build these external
+repos within the .NET source build. See
+[dotnet/source-build](https://github.com/dotnet/source-build) for more details on
+.NET source build.
+
+## How to Build
+
+This repo utilizes the .NET [Arcade](https://github.com/dotnet/arcade) build
+infrastructure. Since this repo is intended solely for source build, it usually
+ makes sense to build with the -sb (source build) flag.
+
+``` bash
+./build.sh -sb
+```
+
+## Adding a New External Component
+
+1. Add the repo as a submodule to `./src`
+
+    ```bash
+    git submodule add <remote_url> ./src/<destination_dir>
+    git commit -m "<commit_message>"
+    ```
+
+1. Define a [project](repo-projects) for the new component. The project
+is responsible building the submodule with the appropriate configuration for
+source build. See the [existing projects](repo-projects) for examples.
+
+1. [Build](#how-to-build) locally and resolve any build errors. Source changes
+must be applied via [patches](patches).
+
+1. Validate the version of the NuGet packages and binaries produced by the build. See the contents of
+`./artifacts/packages/<build_configuration>/NonShipping/Microsoft.SourceBuild.Intermediate.source-build-externals.x.y.z-dev.nupkg`
+
+## Updating an External Component to a Newer Version
+
+1. Update the `src/<external_repo_dir>` to the desired sha
+
+    ``` bash
+    cd src/<external_repo_dir>
+    git checkout <updated_sha>
```

Comment:
I think a `git fetch` is needed here too?

Reply:
Yes you are correct, added.